### ISSUE TITLE
1.17 Server requires Java16 : Debian 10.9

### ISSUE THREAD

**Issue by DparaT** [itzg/docker-minecraft-server#914]
Starting a docker with command: docker run -d -p 25565:25565 --name mc itzg/minecraft-server

results in a:
xxx@sinw:/game# docker logs -f milan2-minecraft
Running as uid=1004 gid=1005 with /data as 'drwxrwxr-x    5 1004     1005          4096 Jun  8 17:12 /data'
Checking version information.
Checking type information.
server.properties already created, skipping
Checking for JSON files.
Copying any mods over...
Setting initial memory to 1G and max to 1G
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: net/minecraft/server/Main has been compiled by a more recent version of the Java Runtime (class file version 6time only recognizes class file versions up to 52.0
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:495)
2021/06/08 17:15:49 Done

Solution: 

Starting a docker with command: docker run -d -p 25565:25565 --name mc itzg/minecraft-server:**java16** results in a running server.
- So don't forget to add the java16 part.

**Comment by itzg** [comment]
Pulling the latest image would have worked too since it also has been updated to Java 16.

**Comment by jdawgzim** [comment]
Pulling the latest didn't seem to work for me.  I'm using Portainer w/ Docker Compose.

Using itzg/minecraft-server:java16 did work for me.

**Comment by CikaRoCC0** [comment]
Doesn't work for me with Docker Compose and - TYPE=SPIGOT

tested with:
itzg/minecraft-server:java16
itzg/minecraft-server:latest

seems, that the container dosn't pull the 1.17 version of SPIGOT
`[init] Downloading Spigot from https://cdn.getbukkit.org/spigot/spigot-1.16.5.jar ...
`

getting this error:
```
[init] Starting the Minecraft server...
Unsupported Java detected (60.0). Only up to Java 15 is supported.
```

Any suggestions are GREATLY appreciated ;)

**Comment by weihao** [comment]
> Doesn't work for me with Docker Compose and - TYPE=SPIGOT
> 
> tested with:
> itzg/minecraft-server:java16
> itzg/minecraft-server:latest
> 
> seems, that the container dosn't pull the 1.17 version of SPIGOT
> `[init] Downloading Spigot from https://cdn.getbukkit.org/spigot/spigot-1.16.5.jar ... `
> 
> getting this error:
> 
> ```
> [init] Starting the Minecraft server...
> Unsupported Java detected (60.0). Only up to Java 15 is supported.
> ```
> 
> Any suggestions are GREATLY appreciated ;)

`docker-compose pull` solves this.

**Comment by dlasher** [comment]
This worked for me to get 1.17 running - :java16 for the win.

**Comment by JonathanShrek** [comment]
> Solution:
> 
> Starting a docker with command: docker run -d -p 25565:25565 --name mc itzg/minecraft-server:**java16** results in a running server.
> 
>     * So don't forget to add the java16 part.

This fixed it for me. Thank you!!

**Comment by stephenbawks** [comment]
Still struggling on this one.  I have an existing container named `mc` and still there, and I really do not want to lose the data in that container.  Pretty sure my son would cry for days.  I am not a docker expert but I cannot seem to get this to run.  

`Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: net/minecraft/server/Main has been compiled by a more recent version of the Java Runtime (class file version 60.0), this version of the Java Runtime only recognizes class file versions up to 52.0`

I pulled the latest image but I am not sure how I tell it how to use latest or update the existing container?

**Comment by itzg** [comment]
@stephenbawks please provide more information. How are you creating the container? Command line args, docker compose file, screenshots?

**Comment by stephenbawks** [comment]
![image](https://user-images.githubusercontent.com/5246651/122407031-52982200-cf4f-11eb-956f-6f148cffaa0f.png)


![image](https://user-images.githubusercontent.com/5246651/122407304-870bde00-cf4f-11eb-846c-23f6c35c9a96.png)

At the moment I am just doing a `docker start mc`

**Comment by JonathanShrek** [comment]
@stephenbawks do you remember how you created the container? For instance the command I use looks like this:

```
docker run \
-d \
-p 31943:25565 \
--name mc \
-e EULA=TRUE \
-e MEMORY=2G \
--restart=always \
-v /home/USER/docker/minecraft/_data:/data \
itzg/minecraft-server:java16
```
***Its important that you do not mess with your volume (-v) data to avoid losing in game data.***

**Comment by stephenbawks** [comment]
I am fairly confident I just used the simple one in the docs

`docker run -d -it -p 25565:25565 -e EULA=TRUE itzg/minecraft-server`

**Comment by itzg** [comment]
@stephenbawks along with re-pulling the latest image, you'll need to make use of this procedure or similar to allow for recreating the container with the data volume carried across

https://github.com/itzg/docker-minecraft-server#converting-anonymous-data-volume-to-named-volume

**Comment by stephenbawks** [comment]
> 
> 
> @stephenbawks along with re-pulling the latest image, you'll need to make use of this procedure or similar to allow for recreating the container with the data volume carried across
> 
> https://github.com/itzg/docker-minecraft-server#converting-anonymous-data-volume-to-named-volume

Okay thank you.  That helped and its back up now.  Just did the data copy to a new volume and started it back up again as a new name.  Now i really should probably back this world up!

**Comment by yllekz** [comment]
Is this updated for multiarch? I'm getting the Java 16 error when using this tag.

**Comment by itzg** [comment]
The java16 and latest tags are already multi-arch

**Comment by xurizaemon** [comment]
I ran into what appears to be the original issue reported here. For me, it was resolved by *updating the version of `itzg/minecraft-server`* (`docker-compose pull mc` for me, or `docker pull itzg/minecraft-server`).

Worth noting that if appending `:java16` appeared to fix this for you, it's plausible that the "fix" was that adding a version tag triggered Docker to retrieve a fresh copy of the image for you.
